catwalk/models/t5.py

```diff
@@ -21,31 +21,35 @@ def get_model(self) -> T5ForConditionalGeneration:
 
     def get_tokenizer(self) -> T5TokenizerFast:
         raise NotImplementedError
+    
+    def _predict_qa(self, task: Task, instances: Sequence[Dict[str, Any]], batch_size: int = 32) -> Iterator[Dict[str, Any]]:
+        qas = MappedSequence(task.instance_conversions[InstanceFormat.HF_QA], instances)
 
-    def predict(  # type: ignore
-        self,
-        task: Task,
-        instances: Sequence[Dict[str, Any]],
-        *,
-        batch_size: int = 32
-    ) -> Iterator[Dict[str, Any]]:
-        if not task.has_instance_conversion(InstanceFormat.T5_PROMPT):
-            raise UnsupportedTaskError(self, task)
+        model = self.get_model().eval()
+        tokenizer = self.get_tokenizer()
+        tokenizer.model_max_length = 1024 # We use this to be similar to the GPT truncation
+
+        with torch.inference_mode():
+            for batch in more_itertools.chunked(Tqdm.tqdm(qas, desc="Processing instances"), batch_size):
+                model_input = tokenizer([f"question:{i.question}" for i in batch],
+                                        [f"context:{i.context}" for i in batch],
```

Comment:
This has question first. GPT2 has context first. Is this on purpose? Same as a reference implementation?

Reply:
T5 has question first in the HuggingFace implementation. There is no reference for which way to format it when fine-tuning GPT models. I was formatting the input in a similar style as in the GPT3 paper prompts. This makes it trivial to run zero-shot inference (just plug in a pre-trained model) and I believe could simplify some logic when we add in-context learning.